src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TypeChar.cs

```diff
@@ -357,12 +372,12 @@ private bool IsFilterCharacter(char ch)
             return GetCompletionRules().IsFilterCharacter(selectedItem, ch, filterText);
         }
 
-        private string GetCurrentFilterText(Model model, CompletionItem selectedItem)
+        private string GetCurrentFilterText(Model model, CompletionItem selectedItem, bool stopBeforeCaret = true)
         {
             var textSnapshot = this.TextView.TextSnapshot;
             var viewSpan = model.GetSubjectBufferFilterSpanInViewBuffer(selectedItem.FilterSpan);
             var filterText = model.GetCurrentTextInSnapshot(
-                viewSpan, textSnapshot, GetCaretPointInViewBuffer());
+                viewSpan, textSnapshot, endPoint: stopBeforeCaret ? (int?)GetCaretPointInViewBuffer() : null);
```

Comment:
I do want to know. In VS2013, if you move the caret to the left changing something then press commit character, you get the identifier replaced, it's useful to correct something:
`Mon|` -> `Mo|n` -> `Mi|n` -> `Mi(|n` -> `Min(|`
In VS2015 you get `Min(|n` in the last step because of this caret check. It's in the code so I thought this change was made intentionally? Personally prefer VS2013 behavior.